hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/CloudObjectsMetaSelector.java

```diff
@@ -0,0 +1,208 @@
+    List<Message> ineligibleMessages = new ArrayList<>();
+
+    ReceiveMessageRequest receiveMessageRequest =
+        new ReceiveMessageRequest()
+            .withQueueUrl(this.queueUrl)
+            .withWaitTimeSeconds(this.longPollWait)
+            .withVisibilityTimeout(this.visibilityTimeout);
+    receiveMessageRequest.setMaxNumberOfMessages(this.maxMessagesEachRequest);
+
+    List<Message> messages =
+        getMessagesToProcess(
+            sqs,
+            this.queueUrl,
+            receiveMessageRequest,
+            this.maxMessageEachBatch,
+            this.maxMessagesEachRequest);
+
+    for (Message message : messages) {
+      boolean isMessageDelete = Boolean.TRUE;
+
+      JSONObject messageBody = new JSONObject(message.getBody());
+      Map<String, Object> messageMap;
+      ObjectMapper mapper = new ObjectMapper();
+
+      if (messageBody.has("Message")) {
+        // If this messages is from S3Event -> SNS -> SQS
+        messageMap =
+            (Map<String, Object>) mapper.readValue(messageBody.getString("Message"), Map.class);
+      } else {
+        // If this messages is from S3Event -> SQS
+        messageMap = (Map<String, Object>) mapper.readValue(messageBody.toString(), Map.class);
+      }
+      if (messageMap.containsKey("Records")) {
+        List<Map<String, Object>> records = (List<Map<String, Object>>) messageMap.get("Records");
+        for (Map<String, Object> record : records) {
+          String eventName = (String) record.get("eventName");
+
+          // filter only allowed s3 event types
+          if (ALLOWED_S3_EVENT_PREFIX.stream().anyMatch(eventName::startsWith)) {
+            eligibleRecords.add(record);
+            isMessageDelete = Boolean.FALSE;
+            processedMessages.add(message);
+
+          } else {
+            log.info("This S3 event " + eventName + " is not allowed, so ignoring it.");
+          }
+        }
+      } else {
+        log.info("Message is not expected format or it's s3:TestEvent");
+      }
+      if (isMessageDelete) {
+        ineligibleMessages.add(message);
+      }
+    }
+    if (!ineligibleMessages.isEmpty()) {
+      deleteBatchOfMessages(sqs, queueUrl, ineligibleMessages);
+    }
+
+    return eligibleRecords;
+  }
+
+  /**
+   * Get the list of events from queue.
+   *
+   * @param sparkContext JavaSparkContext to help parallelize certain operations
+   * @param lastCheckpointStr the last checkpoint time string, empty if first run
+   * @return the list of events
+   */
+  public Pair<List<String>, String> getNextEventsFromQueue(
+      AmazonSQS sqs,
+      JavaSparkContext sparkContext,
+      Option<String> lastCheckpointStr,
+      List<Message> processedMessages) {
+
+    processedMessages.clear();
+
+    log.info("Reading messages....");
+
+    try {
+      log.info("Start Checkpoint : " + lastCheckpointStr);
+
+      long lastCheckpointTime = lastCheckpointStr.map(Long::parseLong).orElse(Long.MIN_VALUE);

```

Comment:
If parsing fails, why set to min_value. negative value does not makes sense. atleast we should set to 0. Or if its epoch, we should set it to the earliest time in epoch (1970/01/01...)